src/classes/BDI_DataImportDeleteBTN_CTRL.cls

```diff
@@ -95,6 +89,18 @@ public with sharing class BDI_DataImportDeleteBTN_CTRL {
         return null;
     }
 
+    private Boolean checkDelete() {
+        return UTIL_Permissions.getInstance().canDelete(DataImport__c.SObjectType);
```

Comment:
Can this call to the `UTIL_Permissions` `canDelete` method be housed within the getter for the local canDelete property above (line 54)?  Then line 139 below could still be `if(canDelete)` which to me seems clearer than having a call to `checkDelete` inside the conditional there.